File: C#/Strategy/Product.cs
using Strategy.Interfaces;

namespace Strategy;

public sealed class Product
{
    public string Name { get; set; }
    public decimal Price { get; set; }
    
    private readonly IPricingStrategy _pricingStrategy;

    public Product(string name, decimal price, IPricingStrategy pricingStrategy)
    {
        Name = name;
        Price = price;
        _pricingStrategy = pricingStrategy;
    }

    /// <summary>
    /// Calculate product price based on <see cref="IPricingStrategy"/> provided
    /// </summary>
    /// <returns>Price after being calculated based on subscription</returns>
    public decimal CalculatePrice()
        => _pricingStrategy.CalculatePrice(Price);
}

File: Python/strategy/Product.py
from PricingStrategy import PricingStrategy

class Product:
    def __init__(self, name: str, price: float, pricing_strategy: PricingStrategy):
        self.name = name
        self.price = price
        self.pricing_strategy = pricing_strategy

    def calculate_price(self) -> float:
        return self.pricing_strategy.calculate_price(self.price)
